Output more detail when ajvOptions.verbose enabled

lvl99 opened the pull request:
This is more an RFC to get people's comments and feedback on how the errors detected during `toMatchSchema` can better help the schema development and test writing process. I used a locally linked version of this to write an extensive schema testing suite and it really helped a lot!

I wanted to have a bit more detail to debug and adjust my schemas during development. Since more detail for errors is available via `ajvOptions.verbose` option I decided to use the value of that to output more detailed messages in the `toMatchSchema` matcher:

* `Received: ...` - displays the value (or schema) that was received that generated the error
* `Expected: ...` - displays the value(s) (or schema) that was expected to not have an error
* `Path: ...` - displays the JSON pointer path to the received value (or schema) that generated the error (also appends the schema's `$id` for convenience)

**TODO:**
- [x] initial working proposal for outputting verbose messages
- [x] resolve test coverage not being 100%
- [x] add documentation detailing the `ajvOptions.verbose` flag and how it influences the `jest-json-schema` error output
- [x] add documentation showing examples of using file-based schemas, along with refs (see issue #7 )

lvl99 commented:
@10xLaCroixDrinker Should be good to review now :+1:

10xLaCroixDrinker commented on README.md:
````diff
@@ -59,6 +59,107 @@ const formats = {
 expect.extend(matchersWithOptions({ formats }));
 ```
 
+### Verbose errors
````
Include a code example of how to enable the verbose logging

lvl99 replied:
Good point! I've added an example now with an inline snapshot for clarity.

anescobar1991 commented:
@lvl99 can you rebase and resolve the conflicts? Once that's done I can merge and we can release!

lvl99 commented:
@anescobar1991 Sure thing!

lvl99 commented:
@anescobar1991 @10xLaCroixDrinker I've rebased and fixed some conflicts. Also updated a few bits here and there for consistency's sake. It should probably necessitate a new review 👍

lvl99 commented:
🎉 

What's the release schedule timeframe for this project on NPM? I noticed that the last official release was 7 months ago.

10xLaCroixDrinker commented:
@lvl99 we can cut a release this week, possibly as early as today

lvl99 commented:
@10xLaCroixDrinker that would be super kind of you if you could! 🙏

10xLaCroixDrinker commented:
@lvl99 v2.1.0 published with the new features you added 👍

lvl99 commented:
Thanks @10xLaCroixDrinker ! Really appreciate the support 🙌
